test/SILGen/switch.swift

```diff
@@ -635,6 +635,133 @@ func test_isa_class_2(x: B) -> AnyObject {
+// CHECK: function_ref @$s6switch39test_isa_pattern_array_downcast_closureyyFySayAA1P_pGcfU_
+// CHECK: } // end sil function '$s6switch39test_isa_pattern_array_downcast_closureyyF'
+func test_isa_pattern_array_downcast_closure() {
+// CHECK-LABEL: sil private [ossa] @$s6switch39test_isa_pattern_array_downcast_closureyyFySayAA1P_pGcfU_ : $@convention(thin) (@guaranteed Array<P>) -> () {
+  let _ = { (ps: [P]) -> Void in
+    // CHECK: bb0(%0 : @guaranteed $Array<P>):
+    switch ps {
+    // CHECK: [[DEST:%[0-9]+]] = alloc_stack $Array<X>
+    // CHECK-NEXT: checked_cast_addr_br copy_on_success Array<P> in [[P:%[0-9]+]] : $*Array<P> to Array<X> in [[DEST]] : $*Array<X>, [[IS_X:bb[0-9]+]], [[IS_NOT_X:bb[0-9]+]]
+    case let _ as [X]:
+      // CHECK: [[IS_X]]:
+      // CHECK-NEXT: load [take] [[DEST]] : $*Array<X>
+      // CHECK: function_ref @$s6switch1ayyF
+      a()
+      // CHECK: br [[CONT:bb[0-9]+]]
+
+    // CHECK: [[IS_NOT_X]]:
+    // CHECK: br [[DEF:bb[0-9]+]]
+    default:
+      // CHECK: [[DEF]]:
+      // CHECK: function_ref @$s6switch1byyF
+      b()
+      // CHECK: br [[CONT]]
+    }
+  }
+  // CHECK: } // end sil function '$s6switch39test_isa_pattern_array_downcast_closureyyFySayAA1P_pGcfU_'
+}
+
+// CHECK-LABEL: sil hidden [ossa] @$s6switch30test_isa_pattern_dict_downcast2psySDySSAA1P_pG_tF : $@convention(thin) (@guaranteed Dictionary<String, P>) -> () {
+func test_isa_pattern_dict_downcast(ps: Dictionary<String, P>) {
+  // CHECK: bb0(%0 : @guaranteed $Dictionary<String, P>):
+  switch ps {
+  // CHECK: checked_cast_addr_br copy_on_success Dictionary<String, P> in [[P:%[0-9]+]] : $*Dictionary<String, P> to Dictionary<String, X> in {{%[0-9]+}} : $*Dictionary<String, X>, [[IS_X:bb[0-9]+]], [[IS_NOT_X:bb[0-9]+]]
+  case is [String : X]:
+    // CHECK: [[IS_X]]:
+    // CHECK: function_ref @$s6switch1ayyF
+    a()
+    // CHECK: br [[CONT:bb[0-9]+]]
+
+  // CHECK: [[IS_NOT_X]]:
+  // CHECK: [[DEST:%[0-9]+]] = alloc_stack $Dictionary<String, Y>
+  // CHECK-NEXT: checked_cast_addr_br copy_on_success Dictionary<String, P> in [[P:%[0-9]+]] : $*Dictionary<String, P> to Dictionary<String, Y> in [[DEST]] : $*Dictionary<String, Y>, [[IS_Y:bb[0-9]+]], [[IS_NOT_Y:bb[0-9]+]]
+  case let _ as [String : Y]:
+    // CHECK: [[IS_Y]]:
+    // CHECK-NEXT: load [take] [[DEST]] : $*Dictionary<String, Y>
+    // CHECK: function_ref @$s6switch1byyF
+    b()
+    // CHECK: br [[CONT]]
+  default:
+    // CHECK: [[IS_NOT_Y]]:
+    // CHECK: function_ref @$s6switch1cyyF
+    c()
+    // CHECK: br [[CONT]]
+  }
+  // CHECK: [[CONT]]:
+  // CHECK: function_ref @$s6switch1dyyF
+  d()
+}
+// CHECK-LABEL: } // end sil function '$s6switch30test_isa_pattern_dict_downcast2psySDySSAA1P_pG_tF'
+
+// CHECK-LABEL: sil hidden [ossa] @$s6switch30test_isa_pattern_set_downcastt2psyShyxG_tSHRzlF : $@convention(thin) <T where T : Hashable> (@guaranteed Set<T>) -> () {
+func test_isa_pattern_set_downcastt<T: Hashable>(ps: Set<T>) {
```

Comment:
Not to commit directly:
```suggestion
// CHECK-LABEL: sil hidden [ossa] @$s6switch29test_isa_pattern_set_downcast2psyShyxG_tSHRzlF : $@convention(thin) <T where T : Hashable> (@guaranteed Set<T>) -> () {
func test_isa_pattern_set_downcast<T: Hashable>(ps: Set<T>) {
```

Reply:
fixed that, thanks!